playwright.config.ts

```diff
@@ -30,6 +30,44 @@ expect.extend({
       pass: false,
     }
   },
+  async toHaveDescription(this: ReturnType<Expect['getState']>, locator: Locator, expected: string) {
+    const page = locator.page()
+    const describedBy = await locator.getAttribute('aria-describedby')
+    const describedByIds = `#${describedBy?.split(' ').join(', #')}`
+
+    const result = page.locator(describedByIds, {
+      hasText: expected,
+    })
+
+    if (await result.isVisible()) {
+      return {
+        message: () => `Expected aria-describedby not to have text content ${this.utils.printReceived(expected)}`,
+        pass: true,
+      }
+    }
+
+    const receivedResult = page.locator(describedByIds)
+    const receivedTextPromises: Promise<string | null>[] = []
+
+    const count = await receivedResult.count()
+    for (let i = 0; i < count; i += 1) {
+      receivedTextPromises.push(receivedResult.nth(i).textContent())
+    }
+
+    const receivedText = await Promise.all(receivedTextPromises)
+
+    return {
+      message: () => `
+        ${
+        this.utils.matcherHint('toHaveDescription', 'HTMLElement', expected, {
+          isNot: this.isNot,
+          promise: this.promise,
+        })
+      }\n
+        Received: ${this.utils.printReceived(receivedText.map((text) => text?.trim()).join('\n'))}`,
```

Comment:
@jpveooys I had to run the trim through a loop here because the loop returns a promise and we have a eslint lint rule that forbids using await inside a loop

Reply:
Looks tidy enough :+1: